addons/mass_mailing/models/mail_message.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, fields
+from odoo.api import depends
+
+class MailMessage(models.Model):
+    _inherit = "mail.message"
+    # ------------------------------------------------------------
+    # ADDITIONS
+    # ------------------------------------------------------------
+
+    mailing_trace_ids = fields.One2many('mailing.trace', 'mail_message_id', compute='_compute_mailing_trace_ids')
+
+    @depends('mail_ids')
+    def _compute_mailing_trace_ids(self):
+        message_trace_mapping = self._get_message_trace_mapping()
+        # browse flattened ids to do a single query
+        traces = self.env['mailing.trace'].browse([trace for traces in message_trace_mapping.values() for trace in traces])
+        for message in self:
+            message.mailing_trace_ids = traces.browse(message_trace_mapping[message.id])
+
+    def _get_message_trace_mapping(self):
+        """Used to allow overrides to the compute method, adding additional results
+            :return Dict[int, mailing_trace_ids]: Dictionary mapping message ids to trace recordsets
+        """
+        # intermediary, no mail value returned
+        mail_mails = self.sudo().mail_ids
+        message_to_mail_trace = {id: [] for id in self.ids}
+        for mail in mail_mails:
+            message_to_mail_trace[mail.mail_message_id.id] += mail.mailing_trace_ids.ids
+        return message_to_mail_trace
```

Comment:
Mm, since I need to link 2 relations of mail_mail I'll need 2 groupby (since this is the method that enables the link between these models..). I think it would add some overhead but I can do it for readability if you think it's worth it.